Packs/Base/Scripts/DBotPreprocessTextData/DBotPreprocessTextData.py

```diff
@@ -84,43 +278,47 @@ def remove_line_breaks(text):
     return re.sub(r"\s+", " ", text.replace("\r", " ").replace("\n", " ")).strip()
 
 
-def hash_word(word, seed):
-    return str(hash_djb2(word, seed))
+def pre_process_batch(data, source_text_field, target_text_field, remove_html_tags, pre_process_type, hash_seed):
+    raw_text_data = [x[source_text_field] for x in data]
+    if remove_html_tags:
+        raw_text_data = [clean_html(x) for x in raw_text_data]
+    raw_text_data = [remove_line_breaks(x) for x in raw_text_data]
+    tokenized_text_data = []
+    for raw_text in raw_text_data:
+        tokenized_text = pre_process_single_text(raw_text, hash_seed, pre_process_type)
+        if hash_seed is None:
+            tokenized_text_data.append(tokenized_text['tokenizedText'])
+        else:
+            tokenized_text_data.append(tokenized_text['hashedTokenizedText'])
+    for d, tokenized_text in zip(data, tokenized_text_data):
+        d[target_text_field] = tokenized_text
+    return data
 
 
-def pre_process(data, source_text_field, target_text_field, remove_html_tags, pre_process_type, hash_seed):
-    tokenized_text_data = [x[source_text_field] for x in data]
-    if remove_html_tags:
-        tokenized_text_data = [clean_html(x) for x in tokenized_text_data]
-    tokenized_text_data = [remove_line_breaks(x) for x in tokenized_text_data]
+def pre_process_single_text(raw_text, hash_seed, pre_process_type):
     pre_process_func = PRE_PROCESS_TYPES[pre_process_type]
-    tokenized_text_data = pre_process_func(tokenized_text_data)
-    for d, tokenized_text_data in zip(data, tokenized_text_data):
-        if hash_seed:
-            tokenized_text_data = " ".join([hash_word(word, hash_seed) for word in tokenized_text_data.split(" ")])
-        d[target_text_field] = tokenized_text_data
-    return data
+    tokenized_text = pre_process_func(raw_text, hash_seed)
+    return tokenized_text
 
 
-def pre_process_nlp(text_data):
-    res = demisto.executeCommand('WordTokenizerNLP', {
-        'value': json.dumps(text_data),
-        'isValueJson': 'yes',
-        'tokenizationMethod': demisto.args()['tokenizationMethod'],
-        'language': demisto.args()['language']
-    })
+def pre_process_nlp(text, seed):
```

Comment:
you can rename this function, it's not nlp anymore, it's more generic - you can get the tokenizer method here as arrgument